src/types/bytearray.rs

```diff
@@ -42,6 +42,48 @@ impl PyByteArray {
         self.len() == 0
     }
 
+    /// Get the start of the buffer containing the contents of the bytearray.
+    ///
+    /// Note that this bytearray object is both shared and mutable, and the backing buffer may be
+    /// reallocated if the bytearray is resized. This can occur from Python code as well as from
+    /// Rust via [PyByteArray::resize].
+    ///
+    /// As a result, the returned pointer should be dereferenced only if since calling this method
+    /// no Python code has executed, [PyByteArray::resize] has not been called.
+    pub fn data(&self) -> *mut u8 {
+        unsafe { ffi::PyByteArray_AsString(self.as_ptr()) as *mut u8 }
+    }
+
+    /// Get the contents of this buffer as a slice.
+    ///
+    /// # Safety
+    /// This is equivalent to `std::slice::from_raw_parts(self.data(), self.len())`, and so all
```

Comment:
This document is well written, but could you please emphasize that **we should not resize or delete the bytearray while holding the slice**?

Reply:
Thanks, tweaked the doc slightly - what do you think of it now?

Reply:
:+1: